tests/phpunit/MediaWiki/Specials/Ask/ErrorWidgetTest.php

```diff
@@ -57,6 +58,18 @@ public function testGetFormattedQueryErrorElement( $errors, $expected ) {
 		);
 	}
 
+	/**
+	 * Return an error box using core MW method
+	 * This is required because the output is different depending on the MW version
+	 *
+	 * @param string $message
+	 * @return $string HTML of the error message box
+	 * @since 5.0.0
+	 */
+	private function getErrorMessageHTML( $message ) {
+		return Html::errorBox( $message, '', 'smw-error-result-error' );
+	}
```

Comment:
Computing the expected value like this is somewhat dodgy. Normally it is good to avoid mirroring the production code in the test.

Reply:
In that case do we want to hardcode the HTML for the error box in the test? We'll need to do multiple versions because it is using Codex in the newer MW version, and Codex components are not stable

Reply:
Well, nvm then. This undermines the value of the tests, but they still seem to provide something.

Reply:
I am not super familiar with PHP tests so I am not sure if it is possible or the right way to do it.
We could validate the HTML/content that we feed into the error box, since the `Html::errorBox` method only provides the wrapper. Would that be a more sensible way to do it?